src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2224,6 +2239,49 @@ export class StudyViewPageStore
         }
     }
 
+    parseRawFilterJson(filterJson: string): any {
+        let rawJson;
+        try {
+            rawJson = JSON.parse(decodeURIComponent(filterJson));
+        } catch (e) {
+            console.error('FilterJson invalid Json: error: ', e);
+        }
+        return rawJson;
+    }
+
+    async fetchSamplesWithStudyIds(studyIds: string[]) {
```

Comment:
you actually don't have to use async/await here.  This function should just return ```internalClient.fetchFilteredSamplesUsingPost(...)```.    You only need to use async/await when you want to do something with the returned data before returning it

Reply:
done